File: 0367-valid-perfect-square/0367-valid-perfect-square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        long long left = 1;
        long long right = num;
        long long ans = 0;
        while(left <= right){
            long long mid = left+ (right-left)/2;
            if(mid * mid >= num){
                ans = mid;
                right = mid - 1;
            }
            else{
                left = mid + 1;
            }
        }
        return(long) ans*ans == num;
    }
};

File: 0367-valid-perfect-square/0367-valid-perfect-square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        lo = 0
        hi = num
        ans = 0
        while lo <= hi:
            mi = lo + (hi - lo) // 2

            if mi * mi >= num:
                ans = mi
                hi = mi - 1
            else:
                lo = mi + 1

        return ans * ans == num


